test/unit/product/product-updater.js

```diff
@@ -34,108 +38,216 @@ describe('Product Updater class', () => {
+        });
+      });
+
+      it('removes vertical and horizontal class from iframe', () => {
+        product.updateConfig(newConfig);
+        assert.calledTwice(removeClassStub);
+        assert.calledWith(removeClassStub.getCall(0), product.classes.product.vertical);
+        assert.calledWith(removeClassStub.getCall(1), product.classes.product.horizontal);
+      });
+
+      it('adds layout class to iframe', () => {
+        product.updateConfig(newConfig);
+        assert.calledOnce(addClassStub);
+        assert.calledWith(addClassStub, product.classes.product.layout);
+      });
+
+      it('resizes view', () => {
+        product.updateConfig(newConfig);
+        assert.calledOnce(resizeStub);
+      });
+    });
+
+    it('adds an event listener on load to resize view for each image in the wrapper', () => {
+      const resizeStub = sinon.stub(product.view, 'resize');
+      const addEventListenerStub = sinon.stub(EventTarget.prototype, 'addEventListener');
+
+      const wrapper = document.createElement('div');
+      const image1 = document.createElement('img');
+      wrapper.appendChild(image1);
+      const image2 = document.createElement('img');
+      wrapper.appendChild(image2);
+      product.view.wrapper = wrapper;
+
+      product.updateConfig(newConfig);
+      assert.calledTwice(addEventListenerStub);
+      assert.calledWith(addEventListenerStub.getCall(0), 'load', sinon.match.func);
+      assert.calledWith(addEventListenerStub.getCall(1), 'load', sinon.match.func);
+
+      addEventListenerStub.getCall(0).args[1]();
+      assert.calledOnce(resizeStub);
+      addEventListenerStub.getCall(1).args[1]();
+      assert.calledTwice(resizeStub);
+
+      resizeStub.restore();
+      addEventListenerStub.restore();
+    });
+
+    it('calls super updateConfig', () => {
+      product.updateConfig(newConfig);
+      assert.calledOnce(superUpdateConfigStub);
+      assert.calledWith(superUpdateConfigStub, newConfig);
+    });
+
+    it('updates config of cart if cart exists', () => {
+      const updateConfigSpy = sinon.spy();
+      product.cart = {
+        updateConfig: updateConfigSpy,
+      };
+      product.updateConfig(newConfig);
+      assert.calledOnce(updateConfigSpy);
+      assert.calledWith(updateConfigSpy, newConfig);
+    });
+
+    it('updates config of modal if modal exists', () => {
+      const updateConfigSpy = sinon.spy();
+      product = Object.defineProperty(product, 'modalProductConfig', {
+        value: {
+          modalProductConfig: 'modalProductConfig',
+        },
+      });
+      product.modal = {
+        updateConfig: updateConfigSpy,
+      };
+      const expectedObject = {
+        options: JSON.parse(JSON.stringify(product.config)),
+      };
+      expectedObject.options.product = product.modalProductConfig;
+
+      product.updateConfig(newConfig);
+      assert.calledOnce(updateConfigSpy);
+      assert.calledWith(updateConfigSpy, expectedObject);
```

Comment:
Is there a way to check that it's also using the config passed into `Product.updateConfig`?